src/lexer.coffee

```diff
@@ -116,10 +117,18 @@ exports.Lexer = class Lexer
     if id is 'from' and @tag() is 'YIELD'
       @token 'FROM', id
       return id.length
-    if id is 'as' and @seenImport and (@tag() is 'IDENTIFIER' or @value() is '*')
-      @tokens[@tokens.length - 1][0] = 'IMPORT_ALL' if @value() is '*'
-      @token 'AS', id
-      return id.length
+    if id is 'as' and @seenImport
+      # loop hack
```

Comment:
Because the code is more selective with the loop. In your case when encountering an `as` after an `{` on an import line, the `as` would always become an `AS` token, which would prevent its currently allowed usage as an import variable.